libnavigation-ui/src/main/java/com/mapbox/navigation/ui/map/NavigationMapboxMap.java

```diff
@@ -553,6 +553,10 @@ public void onStart() {
     handleFpsOnStart();
     locationFpsDelegate.onStart();
 
+    if (navigation != null) {
+      navigation.registerLocationObserver(locationObserver);
+    }
```

Comment:
Is this an issue in the legacy or a regression from when adding support to `1.0` to the UI SDK?

Reply:
I think it's not in legacy. The related logic was added here: https://github.com/mapbox/mapbox-navigation-android/pull/2698/files#diff-e5a5b815f804b7277b6f60237aacc739

Reply:
Wondering if that was not included as part of `onStart` on purpose as we're adding it when on `addProgressChangeListener` https://github.com/mapbox/mapbox-navigation-android/blob/f91471b070aa66c9bbf1736c208ac61d638d3202/libnavigation-ui/src/main/java/com/mapbox/navigation/ui/map/NavigationMapboxMap.java#L345 and then unregistering when on `onStop` assuming that as part of the next `NavigationMapboxMap` lifecycle you need to `addProgressChangeListener`. Also do you know how adding this would affect registering the observer in `addProgressChangeListener`? Are we adding duplicated ones? Could that be a problem 🤔? cc @cafesilencio as you work on https://github.com/mapbox/mapbox-navigation-android/pull/2698

Reply:
@Guardiola31337 - if I understand correctly that you were saying this `locationObserver` might be registered twice, first is in `addProgressChangeListener` sencond is in `onStart`, during the stand-up meeting? That's a good point. 
I verified that it will not because the `mapboxNavigation` is added only after the navigation starts. So I think we should be good to add the `locationObserver` in `onStart()` here.

@cafesilencio , wdyt?

Reply:
> I verified that it will not because the `mapboxNavigation` is added only after the navigation starts.

Ohh I see. So you're saying that `addProgressChangeListener` is now only called once and not as part of the lifecycle. I think that's not the case in the legacy and hence my confusion. If so, yeah, we need to add it as part of the lifecycle so it's re-added when in on configuration changes 👍

Reply:
From what I can remember this was done intentionally in and seems correct.

Reply:
I mean, right now that won't be a problem because of https://github.com/mapbox/mapbox-navigation-android/pull/2834 but if we can avoid one check ¯\\\_(ツ)\_/¯